src/Impostor.Server/Events/Game/Player/PlayerMovementEvent.cs

```diff
@@ -1,24 +1,71 @@
-﻿using Impostor.Api.Events.Player;
+using System;
+using System.Collections.Concurrent;
+
+using Impostor.Api.Events.Player;
 using Impostor.Api.Games;
 using Impostor.Api.Net;
 using Impostor.Api.Net.Inner.Objects;
+using Impostor.Hazel;
 
 namespace Impostor.Server.Events.Player
 {
     // TODO: Finish and use event, needs to be pooled
-    public class PlayerMovementEvent : IPlayerEvent
+    public class PlayerMovementEvent : IPlayerMovementEvent , IRecyclable , IDisposable
     {
+        private static readonly ConcurrentBag<PlayerMovementEvent> items = new ConcurrentBag<PlayerMovementEvent>();
+        private static int counter = 0;
+        private static int MAX = 25;
```

Comment:
A more identifiable name, i.e. `MAX_MOVEMENT_STEPS` or something? (Assuming this is used to hold _X_ amount of previous movements, which could prove quite useful for anticheats.). This could also be a `const` instead of `static`.